src/jit/lower.cpp

```diff
@@ -3737,9 +3746,15 @@ void Lowering::LowerBlock(BasicBlock* block)
     m_block = block;
     m_blockRange = LIR::AsRange(block);
 
-    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(); node != end; node = node->gtNext)
+    for (GenTree* node = m_blockRange.FirstNonPhiNode(), *end = m_blockRange.End(), *next; node != end; node = next)
     {
-        LowerNode(node);
+        node = LowerNode(node);
+
```

Comment:
Okay. I've moved everything to "return the next node to process". I found having the option of returning `nullptr` a bit harder to reason about (esp. in the case that the node being processed is near the end of the block), so I removed that option. Now all of the methods in lowering/decomp that may replace the node being processed simply returns the next node to process.